pipeline_release_process.md

```diff
@@ -0,0 +1,40 @@
+
+<dl>
+  <dt>Switch to the <code>dockstore</code> branch and rebase it on <code>master</code> (or the release branch)</dt>
+  <dd>We want to keep Dockstore-specific detail outside of <code>master</code> so things don't look confusing for people looking at the repo. And, Dockstore allows workflows to be published from specific branches of a repo, so we put everything specific to Dockstore in a branch that we keep up-to-date with released pipeline code.</dd>
+  <dt>Add symlinks to imported WDLs adjacent to the entry point WDL</dt>
+  <dd>We have WDLs that import other WDLs (that import other WDLs...), and we rely on our submission code to handle finding those dependencies and staging them to the imports all resolve correctly. Dockstore has no way of knowing about this though, so we need to insert links into the repo so Dockstore can find the dependencies.
+  Our imports never specify any paths, so we expect all dependency WDLs to be in a flat directory along with the main, entry point WDL. So, we just need to add symlinks to all the imported WDLs alongside the WDLs we're actually going to run. For example:
+  <pre>
+  pipelines/smartseq2_single_sample/SmartSeq2SingleSample.wdl
+  pipelines/smartseq2_single_sample/HISAT.wdl (link to ../../library/tasks/HISAT.wdl)
+  pipelines/smartseq2_single_sample/Picard.wdl (link to ../../library/tasks/Picard.wdl)
+  ...</pre></dd>
+  <dt>Prepare Dockstore test input files</dt>
+  <dd>Our current test input files specify gs urls. One day these will be HCA DSS urls or maybe even GA4GH DOS urls, but that day is not today. Dockstore wants http urls, so we need to prepare a new input file:
+  <pre>
+  {
+    "SmartSeq2SingleCell.gtf_file": "http://storage.googleapis.com/hca-dcp-mint-test-data/reference/GRCh38_Gencode/gencode.v27.primary_assembly.annotation.gtf",
+    "SmartSeq2SingleCell.hisat2_ref_trans_name": "gencode_v27_trans_rsem",
+    "SmartSeq2SingleCell.rrna_intervals": "http://storage.googleapis.com/hca-dcp-mint-test-data/reference/GRCh38_Gencode/gencode.v27.rRNA.interval_list",
+    ...</pre></dd>
+</dl>
+
+Now we're ready to go to [Dockstore](https://dockstore.org/) and actually register the pipeline:
+
+- Go to dockstore.org and login in via github. You'll want to use the github account you use to commit to `skylab`
+- Click on "My Workflows", then "Register Workflow", then enter details about the entry point WDL. The repo name should be `HumanCellAtlas/skylab`, and the workflow path should be something like `/pipelines/smartseq2_single_sample/ss2_single_sample.wdl`.
+- Click "Register Workflow"
+- Select the `dockstore` branch from the dropdown menu and click "Publish"
+- The workflow is now published, but it will be pointing to the most recently updated branch in `skylab`, so it's probably broken :(
```

Comment:
Hey @garyluu @denis-yuen, this is where things kind of fall apart for me. I see that I'm supposed to select a "default" version in the versions tab. But there's also a version dropdown, and it's not clear how those interact or what the "Refresh" button does. Clicking it resets it to the wrong branch, and I never see anything show up in "Files".

Reply:
The versions dropdown is the version you're currently looking at which is what you'll see if the files tab, tools tab, DAG tab, etc.

https://docs.dockstore.org/docs/publisher-tutorials/workflows/ gives a brief description about the refresh button ("You can refresh and publish each repo that you identify as a real workflow in order to get, edit, and display additional information such as available versions.").  I personally only use it when I have a new version or if a file changed on GitHub since the last refresh.

For setting the default branch, it looks like you have the dockstore branch selected:
![image](https://user-images.githubusercontent.com/24548904/41997044-d8b0543a-7a24-11e8-8c53-fc2026c6f42a.png)

As long as the radio button for the default version has the branch selected, it should be fine.  I've created a ticket to investigate why it's bouncing back to a different version after user actions: 
https://github.com/ga4gh/dockstore/issues/1553

Reply:
Thanks Gary. I do see all the correct files now. Is it possible that it takes a little time for the files to update?

Reply:
It shouldn't, my guess is that a different branch version was displayed